packages/office-ui-fabric-react/src/components/TextField/TextField.scss

```diff
@@ -155,6 +155,11 @@
   font-size: $ms-font-size-xs;
 }
 
+.root.rootIsBorderless {
```

Comment:
We should probably be changing the color to transparent rather than nixing the border width. Removing the border might affect alignment and/or high contrast borders.